### PR TITLE
feat: Added video streaming latency check tool

### PR DESCRIPTION
**Pull request by karasusan**
- Added the tool for measuring latency of video streaming
- Sender encodes timestamps into textures each frames and receiver decodes them
- The codec is very simple that it transform 32bit integer to 2D-barcode

Note: 
This tool is not supported on Android because this uses Compute Shader.

https://user-images.githubusercontent.com/1132081/149113935-95e287ed-5d38-434a-8a8d-4192bdb3ed7d.mp4

### CODE REVIEW

File: Samples~/E2ELatency/BarcodeDecoder.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        return Decode(texture);
+    }
+
+    int Decode(Texture source)
+    {
+        Shader.SetTexture(0, "Source", source);
+        Shader.SetInt("Row", Row);
+        Shader.SetInt("Column", Column);
+        Shader.SetBuffer(0, "Result", readbackBuffer_);
+        Shader.Dispatch(0, Row, Column, 1);
```
Comment (kannan-xiao4) [inline Samples~/E2ELatency/BarcodeDecoder.cs:37]: Isn't it necessary to use the result of `Shader.FindKernelIndex`?

File: Samples~/Scripts/SceneSelectUI.cs [hunk 2]
```diff
@@ -67,6 +68,11 @@ void Start()
             buttonVideoReceive.onClick.AddListener(OnPressedVideoReceiveButton);
             buttonBandwidth.onClick.AddListener(OnPressedBandwidthButton);
             buttonPerfectNegotiation.onClick.AddListener(OnPressedPerfectNegotiationButton);
+            buttonLatency.onClick.AddListener(OnPressedLatencyButton);
+
+            // This sample uses Compute Shader, so almost Android devices don't work correctly.
```
[the remark this answers is not in the document]
Comment (karasusan) [reply Samples~/Scripts/SceneSelectUI.cs:73]: nice idea!

[the remark this answers is not in the document]
Comment (karasusan) [reply Samples~/Scripts/SceneSelectUI.cs:73]: fixed it

File: Samples~/E2ELatency/BarcodeDecoder.cs (hunk 1 above)
Comment (karasusan) [reply Samples~/E2ELatency/BarcodeDecoder.cs]: fixed it

**Review by kannan-xiao4 — APPROVED** [review]

**Comment by ZeoWorks** [comment]
Great tool Karasusan!
How is progress in terms of resolving the standalone decoding latency issue?

**Comment by karasusan** [comment]
@ZeoWorks
I fixed for decoding latency issue that moving the flipping image process here.
https://github.com/Unity-Technologies/com.unity.webrtc/pull/591

**Comment by karasusan** [comment]
@ZeoWorks
Also, framerate affects results of measuring. We should measure latency with high framerate for getting accreting results.

**Comment by ZeoWorks** [comment]
> @ZeoWorks I fixed for decoding latency issue that moving the flipping image process here. #591

Thank you, I noticed you mentioned that it decreases delay by 1 frame (16ms at 60fps), awesome!
However, there is still a pretty high delay as you mentioned in 'issues';
Win Unity Editor -> Chrome M96: 83ms
Win Unity Editor -> Win Runtime: 149ms
Win Unity Editor -> Win Runtime: With this Patch 136ms

Ideally, we should measure latency with the in-game framerate capped to the same frame rate of the encoding.
(60fps in-game for encoder, 60fps in-game for decoder).

**Pull request merged**